File: core/src/main/java/com/liveOne/github/botDS/configuration/MConfig.java
package com.liveOne.github.botDS.configuration;

import dev.dejvokep.boostedyaml.YamlDocument;
import java.io.File;
import java.io.IOException;
import java.util.Objects;

public class MConfig {

    public static YamlDocument config;
    static {
        try {
          config = YamlDocument.create(new File("config.yml"), Objects.requireNonNull(MConfig.class.getClassLoader().getResourceAsStream("config.yml")));
        } catch (IOException e) {throw new RuntimeException(e);}
    }


    public static void saveConfiguration() {
        try {
            config.save();
        } catch (Exception e) {System.out.printf("Failed to save file %s", e.getCause());}}

}

File: src/main/kotlin/configuration/MConfig.kt
package com.liveOne.github.botDS.configuration

import dev.dejvokep.boostedyaml.YamlDocument
import java.io.*

object MConfig {
    val config: YamlDocument = YamlDocument.create(File("config.yml"), javaClass.classLoader.getResourceAsStream("config.yml")!!)


    fun saveConfiguration() {
        try {
            config.save()
        } catch (e: Exception) {System.out.printf("Failed to save file %s", e.cause)} }
}

